apollo-compiler/src/test/graphql/com/example/simple_inline_fragment/TestQuery.java

```diff
@@ -245,34 +245,26 @@ public AsDroid map(ResponseReader reader) throws IOException {
     }
 
     public static class Hero {
-      private final @Nonnull String name;
-
       private final Optional<AsHuman> asHuman;
 
       private final Optional<AsDroid> asDroid;
 
-      public Hero(@Nonnull String name, @Nullable AsHuman asHuman, @Nullable AsDroid asDroid) {
-        this.name = name;
-        this.asHuman = Optional.fromNullable(asHuman);
-        this.asDroid = Optional.fromNullable(asDroid);
-      }
-
-      public @Nonnull String name() {
-        return this.name;
-      }
-
-      public Optional<AsHuman> asHuman() {
-        return this.asHuman;
-      }
-
-      public Optional<AsDroid> asDroid() {
-        return this.asDroid;
+      public Hero() {
+        if (this instanceof AsHuman) {
```

Comment:
We don't support multiple inline fragments (see comment below).
And codegen merges all fields in IR for inline fragments. So you will have access to `name` field as it will be merged into `Plant` fragment